File: design-pattern-java/app/src/test/java/design/pattern/java/patterns/structural/bridge/BridgeTest.java
package design.pattern.java.patterns.structural.bridge;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class BridgeTest {
    class SomeLibraryMock extends SomeLibrary {
        @Override
        public String writeString() {
            return "writeString";
        }
        @Override
        public String writeOtherString() {
            return "writeOtherString";
        }
        @Override
        public Integer integerGenerator() {
            return 5;
        }
    }
    @Test
    public void shouldCallLibraryMethodsWhenBridgeFeatureIsCalled() {
        SomeLibrary someLibraryMock = new SomeLibraryMock();
        ServiceBridge bridge = new ServiceBridge(someLibraryMock);
        String result = bridge.concreteFeature();
        Assertions.assertTrue(
            result.contains(someLibraryMock.writeString()));
        Assertions.assertTrue(
            result.contains(someLibraryMock.writeOtherString()));
        Assertions.assertTrue(
            result.contains(someLibraryMock.integerGenerator().toString()));
    }
}

File: design-pattern-jvm/app/src/test/kotlin/design/pattern/kotlin/patterns/structural/bridge/BridgeTest.kt
package design.pattern.kotlin.patterns.structural.bridge

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class BridgeTestKt {
    internal inner class SomeLibraryMock : SomeLibrary() {
        override fun writeString(): String {
            return "writeString"
        }

        override fun writeOtherString(): String {
            return "writeOtherString"
        }

        override fun integerGenerator(): Int {
            return 5
        }
    }

    @Test
    fun shouldCallLibraryMethodsWhenBridgeFeatureIsCalled() {
        val someLibraryMock: SomeLibrary = SomeLibraryMock()
        val bridge = ServiceBridge(someLibraryMock)
        val result = bridge.concreteFeature()
        Assertions.assertTrue(result.contains(someLibraryMock.writeString()))
        Assertions.assertTrue(result.contains(someLibraryMock.writeOtherString()))
        Assertions.assertTrue(result.contains(someLibraryMock.integerGenerator().toString()))
    }
}
